spec/models/defect_spec.rb

```diff
@@ -124,6 +124,20 @@
     end
   end
 
+  describe '#reference_number' do
+    it 'returns a formatted identifier code' do
+      defect = create(:defect, sequence_number: 42)
+      expect(defect.reference_number).to eq('NB-000-042')
+    end
+
+    it 'uses sequential identifiers' do
+      defects = 5.times.map { create(:defect) }
+      ids = defects.map { |d| d.reference_number.scan(/\d/).join('').to_i }
```

Comment:
I think you could achieve the same with `ids = defects.pluck(:sequence_number)`

Reply:
This would have the same effect, but I specifically want to test the `reference_number` method as that's what's displayed to the user, rather than testing the implementation.